src/Autofac.Multitenant/MultitenantContainer.cs

```diff
@@ -657,5 +658,32 @@ protected override void Dispose(bool disposing)
 
             base.Dispose(disposing);
         }
+
+        /// <inheritdoc />
+        protected override async ValueTask DisposeAsync(bool disposing)
+        {
+            if (disposing)
+            {
+                _readWriteLock.EnterWriteLock();
+
+                try
+                {
+                    foreach (var scope in _tenantLifetimeScopes.Values)
+                    {
+                        await scope.DisposeAsync();
+                    }
+
+                    await this.ApplicationContainer.DisposeAsync();
+                }
+                finally
+                {
+                    _readWriteLock.ExitWriteLock();
+                }
+
+                _readWriteLock.Dispose();
```

Comment:
Should this Dispose be inside the finally block? Otherwise an exception in one of the nested dispose calls won't dispose the lock. I realise that the existing dispose doesn't do this though...

Reply:
I applied the changes for both `Dispose` methods.